volumina/widgets/exportHelper.py

```diff
@@ -35,63 +34,110 @@
-### lazyflow
-###
-_has_lazyflow = True
-try:
-    from lazyflow.graph import Graph
-    from lazyflow.request import Request
-    from lazyflow.utility import PathComponents
-except ImportError as e:
-    exceptStr = str(e)
-    _has_lazyflow = False
+import lazyflow
+from lazyflow.request import Request
+from lazyflow.utility import PathComponents
+from lazyflow.operators import OpMultiArrayStacker
+from lazyflow.operators.ioOperators import OpFormattedDataExport
 
 
-def get_settings_and_export_layer(layer, parent_widget=None):
-    """
-    Prompt the user for layer export settings, and perform the layer export.
+logger = logging.getLogger(__name__)
+
+
+def _get_export_slots(layer: Layer) -> typing.List[lazyflow.slot.Slot]:
+    """Returns export slots for layer
+
+    Args:
+        layer (Layer): Volumina Layer, its datasources will be checked for "dataSlot" member.
+
+    Returns:
+        List of slots that can be exported from layer
+
+    Raises:
+        RuntimeError: will be raised if the is no dataSource that has the "dataSlot" attribute, aka
+          that is a lazyflow dataSource (slot)
     """
-    sourceTags = [True for l in layer.datasources]
-    for i, source in enumerate(layer.datasources):
-        if not hasattr(source, "dataSlot"):
-            sourceTags[i] = False
+    sourceTags = [
+        hasattr(l, "dataSlot") and l.dataSlot is not None for l in layer.datasources
+    ]
     if not any(sourceTags):
         raise RuntimeError(
             "can not export from a non-lazyflow data source (layer=%r, datasource=%r)"
             % (type(layer), type(layer.datasources[0]))
         )
+    return [
+        slot.dataSlot
+        for (slot, isSlot) in zip(layer.datasources, sourceTags)
+        if isSlot is True
+    ]
 
-    if not _has_lazyflow:
-        raise RuntimeError("lazyflow not installed")
-    import lazyflow
 
-    dataSlots = [slot.dataSlot for (slot, isSlot) in zip(layer.datasources, sourceTags) if isSlot is True]
+def _get_stacked_data_sources(layer: Layer) -> OpMultiArrayStacker:
+    """Get operator with stacked output from all slots of layer
+
+    Args:
+        layer (Layer): layer containing datasources that can be exported
+          (datasource.dataSlot -> Slot)
+
+    Returns:
+        OpMultiArrayStacker: Operator configured with all stackable input slots
+        from layer.
+    """
+    dataSlots = _get_export_slots(layer)
 
-    opStackChannels = lazyflow.operators.OpMultiArrayStacker(dataSlots[0].getRealOperator().parent)
+    opStackChannels = lazyflow.operators.OpMultiArrayStacker(
+        dataSlots[0].getRealOperator().parent
+    )
     for slot in dataSlots:
-        assert isinstance(slot, lazyflow.graph.Slot), "slot is of type %r" % (type(slot))
-        assert isinstance(slot.getRealOperator(), lazyflow.graph.Operator), "slot's operator is of type %r" % (
-            type(slot.getRealOperator())
+        assert isinstance(slot, lazyflow.graph.Slot), "slot is of type %r" % (
```

Comment:
Maybe use new and shiny f-strings here and in other changes in this PR?

Reply:
yeh, made sure all such formats are with f-stings now! thx